pkg/controllers/clusterresourceplacement/suite_test.go

```diff
@@ -0,0 +1,183 @@
+	Expect(err).Should(Succeed())
+
+	//+kubebuilder:scaffold:scheme
+	By("construct the k8s client")
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	Expect(err).Should(Succeed())
+	Expect(k8sClient).NotTo(BeNil())
+
+	dynamicClient, err := dynamic.NewForConfig(cfg)
+	Expect(err).Should(Succeed())
+	Expect(dynamicClient).NotTo(BeNil())
+
+	By("starting the controller manager")
+	klog.InitFlags(flag.CommandLine)
+	flag.Parse()
+
+	mgr, err = ctrl.NewManager(cfg, ctrl.Options{
+		Scheme:             scheme.Scheme,
+		MetricsBindAddress: "0",
+		Logger:             klogr.NewWithOptions(klogr.WithFormat(klogr.FormatKlog)),
+		Port:               4848,
+	})
+	Expect(err).Should(Succeed(), "failed to create manager")
+
+	reconciler := &Reconciler{
+		Client:                 mgr.GetClient(),
+		Scheme:                 mgr.GetScheme(),
+		UncachedReader:         mgr.GetAPIReader(),
+		Recorder:               mgr.GetEventRecorderFor(controllerName),
+		RestMapper:             mgr.GetRESTMapper(),
+		InformerManager:        informer.NewInformerManager(dynamicClient, 5*time.Minute, ctx.Done()),
+		DisabledResourceConfig: utils.NewDisabledResourceConfig(),
+		SkippedNamespaces: map[string]bool{
+			"default": true,
+		},
+	}
+	opts := options.RateLimitOptions{
+		RateLimiterBaseDelay:  5 * time.Millisecond,
+		RateLimiterMaxDelay:   60 * time.Second,
+		RateLimiterQPS:        10,
+		RateLimiterBucketSize: 100,
+	}
+	rateLimiter := options.DefaultControllerRateLimiter(opts)
+	crpController := controller.NewController(controllerName, controller.NamespaceKeyFunc, reconciler.Reconcile, rateLimiter)
+
+	// Set up the watchers
+	err = (&clusterschedulingpolicysnapshot.Reconciler{
+		Client:              mgr.GetClient(),
+		PlacementController: crpController,
+	}).SetupWithManager(mgr)
+	Expect(err).Should(Succeed(), "failed to create clusterSchedulingPolicySnapshot watcher")
+
+	err = (&clusterresourceplacementwatcher.Reconciler{
+		PlacementController: crpController,
+	}).SetupWithManager(mgr)
+	Expect(err).Should(Succeed(), "failed to create clusterResourcePlacement watcher")
+
+	ctx, cancel = context.WithCancel(context.TODO())
+	// Run the controller manager
+	go func() {
+		defer GinkgoRecover()
+		err := mgr.Start(ctx)
+		Expect(err).Should(Succeed(), "failed to run manager")
+	}()
+	// Run the crp controller
+	go func() {
+		err := crpController.Run(ctx, 1)
+		Expect(err).Should(Succeed(), "failed to run crp controller")
+	}()
+
+	By("By creating member clusters namespaces")
+	ns := corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: member1Namespace,
+		},
+	}
+	Expect(k8sClient.Create(ctx, &ns)).Should(Succeed())
+
+	ns = corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: member2Namespace,
+		},
+	}
+	Expect(k8sClient.Create(ctx, &ns)).Should(Succeed())
+})
+
+var _ = AfterSuite(func() {
+	defer klog.Flush()
+
+	By("By deleting member clusters namespaces")
+	ns := corev1.Namespace{
```

Comment:
Is this cleanup really necessary since the test environment itself is being teared apart?